pkg/service/process_manager.go

```diff
@@ -77,8 +78,27 @@ func (s *ProcessManager) startIngress(ctx context.Context, p *params.Params, clo
 	defer span.End()
 
 	h := &process{
-		info:          p.IngressInfo,
-		sessionCloser: closeSession,
+		info: p.IngressInfo,
+		sessionCloser: func(ctx context.Context) {
+			if closeSession != nil {
+				closeSession(ctx)
+			}
+
+			go func() {
+				time.Sleep(time.Second)
+
+				s.mu.Lock()
+				h := s.activeHandlers[p.State.ResourceId]
```

Comment:
Is it possible to unlock as soon as `h` is returned from the map?

Reply:
Sure